DXRModules/DeusEx/Classes/DXRBingoCampaign.uc

```diff
@@ -0,0 +1,257 @@
+class DXRBingoCampaign extends DXRActorsBase transient;
+
+function PreFirstEntry()
+{
+    local DXREvents events;
+    local BlackHelicopter jock;
+    local Switch1 button;
+    local PayPhone invisibleWall;
+    local #var(prefix)FlagTrigger ft;
+
+    if (!dxr.flags.IsBingoCampaignMode()) return;
+    Super.PreFirstEntry();
+
+    foreach AllActors(class'DXREvents', events) {
+        switch (dxr.localURL) {
+            case "02_NYC_BATTERYPARK":
+                dxr.flags.settings.starting_map = 20;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "03_NYC_UNATCOISLAND":
+                dxr.flags.settings.starting_map = 30;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "03_NYC_AIRFIELD":
+                foreach AllActors(class'BlackHelicopter', jock) {
+                    AddBingoEventBlocker(jock, 'DXRando_Mission03_BingoCompleted');
+                    break;
+                }
+                break;
+            case "04_NYC_UNATCOISLAND":
+                dxr.flags.settings.starting_map = 40;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "05_NYC_UNATCOMJ12LAB":
+                if (!dxr.flagbase.GetBool('DXRando_Mission04_BingoCompleted')) {
+                    SetTimer(1.0, true);
```

Comment:
If Gunther's conversation is blocked (see later comment), you could likely avoid this being necessary by doing something to prevent the mission script from taking you to the next mission when you die (so you just die normally instead)

Reply:
Trust me, I tried spent quite a bit of time trying multiple ideas.

Reply:
We already inject on the Mission04 mission script - you should just need to not call the Super.Timer if the bingo line isn't completed (and obviously not do the stuff to send you to the next map)

Reply:
It works when done inside `DXRMission04`.